compiler/rustc_infer/src/infer/outlives/test_type_match.rs

```diff
@@ -137,7 +137,7 @@ impl<'tcx> TypeRelation<TyCtxt<'tcx>> for MatchAgainstHigherRankedOutlives<'tcx>
         "MatchAgainstHigherRankedOutlives"
     }
 
-    fn tcx(&self) -> TyCtxt<'tcx> {
+    fn cx(&self) -> TyCtxt<'tcx> {
```

Comment:
Uh... is there broader context for these changes I'm unaware of? As written it seems to that having a `tcx` getter named `cx` (and all the similar changes) makes the code less clear.

Reply:
tcx and TyCtxt mean nothing in rustc_type_ir, and the name interner is quite long.

@lcnr and I decided it's idiomatic enough just to call it cx in the abstracted-out new solver crate.

Reply:
That makes sense for the changes `rustc_type_ir`, but this change is in `rustc_infer` where `TyCtxt` does mean something. It's the return type of this method! And there are also changes in `rustc_borrowck`, `rustc_middle`, etc.

Reply:
Yes, And the trait definition is in `rustc_type_ir`. Is this affecting your ability to understand the code? I feel like this can be fixed with documentation.

Reply:
Like, OK, we have more than a handful of other random strings that people are going to be confused about once, learn, and then never think about forever. This particular one is in `TypeRelation`, which approximately nobody should ever touch.

Reply:
Also like, `TyCtxt` really *is* the `Cx` of all `Cx`es. If anything deserves that name, at least in a somewhat limited context of things that touch the `middle::ty` layer, then it's definitely this.

Reply:
The trait is in `rustc_type_ir`, hmm, ok. You are right that `interner` would definitely be a better name, though I understand the desire for a short name. Thanks for the explanation.